geniza/corpus/views.py

```diff
@@ -752,13 +753,24 @@ def get_context_data(self, **kwargs):
         except Source.DoesNotExist:
             raise Http404
 
-        # if we have neither IIIF images nor transcription content with placeholder canvases,
-        # pass two placeholder canvases for use in editor
-        if not context_data["images"]:
-            canvas_base_uri = "%siiif/canvas/" % self.get_object().permalink
-            for i in [1, 2]:
-                canvas_uri = "%s%d/" % (canvas_base_uri, i)
-                context_data["images"][canvas_uri] = Document.PLACEHOLDER_CANVAS
+        # per each fragment without images, pass two placeholder canvases for use in editor
+        for b in self.object.textblock_set.all():
+            frag_images = b.fragment.iiif_images()
+            if frag_images is None or not frag_images[0]:
```

Comment:
why do you need to check `frag_images[0]` ?

Reply:
It's a very rare case, but it is possible that `fragment.iiif_images()` could return the list `[[], [], []]`. It would happen when `fragment.iiif_url` is defined but `fragment.manifest` is not, and then there is an error retrieving the remote manifest. It happens in testing, which is why I added it!

Reply:
Ah. is it an error in the other code that we're returning that? 

Please add a brief comment here to document this rare case, just so we know why/when it could happen.

Would it help if we got rid of the remote manifest loading? I remember we planned to phase that out but didn't finish all the refactoring. It makes sense to me it could happen in test; I hope it won't happen in production, or at least only very rarely.

Reply:
Good point, it probably makes more sense to just `return None` in that case from the other code.

> Would it help if we got rid of the remote manifest loading?
Yes, but we'll need to finish up the refactor as you say! I'm not sure where it falls in priority.

Reply:
> Yes, but we'll need to finish up the refactor as you say! I'm not sure where it falls in priority.

low priority — not your job unless it starts impacting the site either for feature development or is related to a bug (seems unlikely)